Test runner fails with -cc g++

### Describe the bug

`v test` fails when using `-cc g++` .
The compilation with g++ backnd works fine.

Full command:

```bash
v -cc g++ -cflags -std=c++20 test <module>
```

Compiler versions:

```bash
~> g++ --version 
g++ (Ubuntu 11.4.0-1ubuntu1~22.04) 11.4.0
```

I tried with g++-10, g++-11, the result is the same:

```bash
>> compilation failed:
/tmp/v_1000/tsession_779ea87fd740_01JKX4J5P4901PR1YTTSFM9EZ5/runner_test.01JKX4J75RKHPDPD0S55VK6KKH.tmp.c:2949:1: error: designator order for field ‘_main__TestRunner_interface_methods::_method_start’ does not match declaration order in ‘_main__TestRunner_interface_methods’
 2949 | };
      | ^
```

My guess is that in the compiled source  `_method__v_free` of `_main__TestRunner_interface_methods`  is out of place. Line 2926 on the screenshot.

![Image](https://github.com/user-attachments/assets/01b63ac4-01e3-4ba3-a3d3-2dc888d8b7a3)

### Reproduction Steps

`my/my_test.v`:

```v
module my

fn test_version() {
	assert 1 == 1
}
```

Run:

```bash
v -cc g++ test my
```

### Expected Behavior

```
---- Testing... ----------------------------------------------------------------------------------------------------------------------------
 OK       3.217 ms ...
```

### Current Behavior

```
>> compilation failed:
/tmp/v_1000/tsession_779ea87fd740_01JKX4J5P4901PR1YTTSFM9EZ5/runner_test.01JKX4J75RKHPDPD0S55VK6KKH.tmp.c:2949:1: error: designator order for field ‘_main__TestRunner_interface_methods::_method_start’ does not match declaration order in ‘_main__TestRunner_interface_methods’
 2949 | };
```

### Possible Solution

_No response_

### Additional Information/Context

_No response_

### V version

0.4.9 d1524fe

### Environment details (OS name and version, etc.)

Linux x86_64

```bash
~> uname -a                                                                                                                                                    git:(master|…6
Linux workhorse 6.9.3-76060903-generic #202405300957~1738770968~22.04~d5f7c84 SMP PREEMPT_DYNAMIC Wed F x86_64 x86_64 x86_64 GNU/Linux
```

I haven't tried on Mac yet, but i think g++ will be the same version.

> [!NOTE]
> You can use the 👍 reaction to increase the issue's priority for developers.
>
> Please note that only the 👍 reaction to the issue itself counts as a vote.
> Other reactions and those to comments will not be taken into account.

--- a/vlib/v/gen/c/cgen.v
+++ b/vlib/v/gen/c/cgen.v
@@ -7533,7 +7533,8 @@ fn (mut g Gen) interface_table() string {
 		methods_struct_name := 'struct _${interface_name}_interface_methods'
 		mut methods_struct_def := strings.new_builder(100)
 		methods_struct_def.writeln('${methods_struct_name} {')
-		inter_methods := inter_info.get_methods()
+		mut inter_methods := inter_info.get_methods()
+		inter_methods.sort(a < b)
 		mut methodidx := map[string]int{}
 		for k, method_name in inter_methods {
 			method := isym.find_method_with_generic_parent(method_name) or { continue }
@@ -7664,7 +7665,9 @@ static inline __shared__${interface_name} ${shared_fn_name}(__shared__${cctype}*
 				methods_struct.writeln('\t{')
 			}
 			if st == ast.voidptr_type || st == ast.nil_type {
-				for mname, _ in methodidx {
+				mut mnames := methodidx.keys()
+				mnames.sort(a < b)
+				for mname in mnames {
 					if g.pref.build_mode != .build_module {
 						methods_struct.writeln('\t\t._method_${c_fn_name(mname)} = (void*) 0,')
 					}
@@ -7721,7 +7724,9 @@ static inline __shared__${interface_name} ${shared_fn_name}(__shared__${cctype}*
 				}
 			}
 
-			for method in methods {
+			mut ordered_methods := methods.clone()
+			ordered_methods.sort(a.name < b.name)
+			for method in ordered_methods {
 				mut name := method.name
 				if method.generic_names.len > 0 && inter_info.parent_type.has_flag(.generic) {
 					parent_sym := g.table.sym(inter_info.parent_type)

--- a/vlib/v/gen/c/testdata/iface_method_order.c.must_have
+++ b/vlib/v/gen/c/testdata/iface_method_order.c.must_have
@@ -1,0 +1,18 @@
+struct _main__Foo_interface_methods {
+void (*_method_a)(void* _);
+void (*_method_b)(void* _);
+};
+struct _main__Foo_interface_methods main__Foo_name_table[3] = {
+{
+._method_a = (void*) main__Bar_a_Interface_main__Foo_method_wrapper,
+._method_b = (void*) main__Bar_b_Interface_main__Foo_method_wrapper,
+},
+{
+._method_a = (void*) 0,
+._method_b = (void*) 0,
+},
+{
+._method_a = (void*) main__Baz_a_Interface_main__Foo_method_wrapper,
+._method_b = (void*) main__Baz_b_Interface_main__Foo_method_wrapper,
+},
+};

--- a/vlib/v/gen/c/testdata/iface_method_order.vv
+++ b/vlib/v/gen/c/testdata/iface_method_order.vv
@@ -1,0 +1,18 @@
+interface Foo {
+	a()
+	b()
+}
+
+struct Bar implements Foo {
+}
+
+fn (b Bar) b() {}
+
+fn (b Bar) a() {}
+
+struct Baz implements Foo {
+}
+
+fn (b Baz) b() {}
+
+fn (b Baz) a() {}

